// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/MonthlyUsageAttributionSupportedMetrics.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PROFILED_CONTAINER_PERCENTAGE = exports.PROFILED_CONTAINER_USAGE = exports.NPM_HOST_PERCENTAGE = exports.NPM_HOST_USAGE = exports.INVOCATIONS_PERCENTAGE = exports.INVOCATIONS_USAGE = exports.INFRA_HOST_PERCENTAGE = exports.INFRA_HOST_USAGE = exports.FUNCTIONS_PERCENTAGE = exports.FUNCTIONS_USAGE = exports.FARGATE_PERCENTAGE = exports.FARGATE_USAGE = exports.ESTIMATED_INGESTED_SPANS_PERCENTAGE = exports.ESTIMATED_INGESTED_SPANS_USAGE = exports.ESTIMATED_INDEXED_SPANS_PERCENTAGE = exports.ESTIMATED_INDEXED_SPANS_USAGE = exports.ESTIMATED_INGESTED_LOGS_PERCENTAGE = exports.ESTIMATED_INGESTED_LOGS_USAGE = exports.ESTIMATED_INDEXED_LOGS_PERCENTAGE = exports.ESTIMATED_INDEXED_LOGS_USAGE = exports.DBM_QUERIES_USAGE = exports.DBM_QUERIES_PERCENTAGE = exports.DBM_HOSTS_USAGE = exports.DBM_HOSTS_PERCENTAGE = exports.CWS_HOSTS_USAGE = exports.CWS_HOSTS_PERCENTAGE = exports.CWS_CONTAINERS_USAGE = exports.CWS_CONTAINERS_PERCENTAGE = exports.CUSTOM_TIMESERIES_PERCENTAGE = exports.CUSTOM_TIMESERIES_USAGE = exports.CSPM_HOSTS_USAGE = exports.CSPM_HOSTS_PERCENTAGE = exports.CSPM_CONTAINERS_USAGE = exports.CSPM_CONTAINERS_PERCENTAGE = exports.CONTAINER_PERCENTAGE = exports.CONTAINER_USAGE = exports.CONTAINER_EXCL_AGENT_PERCENTAGE = exports.CONTAINER_EXCL_AGENT_USAGE = exports.BROWSER_PERCENTAGE = exports.BROWSER_USAGE = exports.APPSEC_PERCENTAGE = exports.APPSEC_USAGE = exports.APM_HOST_PERCENTAGE = exports.APM_HOST_USAGE = exports.APPSEC_FARGATE_PERCENTAGE = exports.APPSEC_FARGATE_USAGE = exports.APM_FARGATE_PERCENTAGE = exports.APM_FARGATE_USAGE = exports.API_PERCENTAGE = exports.API_USAGE = void 0;
exports.ALL = exports.ESTIMATED_RUM_SESSIONS_PERCENTAGE = exports.ESTIMATED_RUM_SESSIONS_USAGE = exports.SNMP_PERCENTAGE = exports.SNMP_USAGE = exports.PROFILED_HOST_PERCENTAGE = exports.PROFILED_HOST_USAGE = exports.PROFILED_FARGATE_PERCENTAGE = exports.PROFILED_FARGATE_USAGE = void 0;
exports.API_USAGE = "api_usage";
exports.API_PERCENTAGE = "api_percentage";
exports.APM_FARGATE_USAGE = "apm_fargate_usage";
exports.APM_FARGATE_PERCENTAGE = "apm_fargate_percentage";
exports.APPSEC_FARGATE_USAGE = "appsec_fargate_usage";
exports.APPSEC_FARGATE_PERCENTAGE = "appsec_fargate_percentage";
exports.APM_HOST_USAGE = "apm_host_usage";
exports.APM_HOST_PERCENTAGE = "apm_host_percentage";
exports.APPSEC_USAGE = "appsec_usage";
exports.APPSEC_PERCENTAGE = "appsec_percentage";
exports.BROWSER_USAGE = "browser_usage";
exports.BROWSER_PERCENTAGE = "browser_percentage";
exports.CONTAINER_EXCL_AGENT_USAGE = "container_excl_agent_usage";
exports.CONTAINER_EXCL_AGENT_PERCENTAGE = "container_excl_agent_percentage";
exports.CONTAINER_USAGE = "container_usage";
exports.CONTAINER_PERCENTAGE = "container_percentage";
exports.CSPM_CONTAINERS_PERCENTAGE = "cspm_containers_percentage";
exports.CSPM_CONTAINERS_USAGE = "cspm_containers_usage";
exports.CSPM_HOSTS_PERCENTAGE = "cspm_hosts_percentage";
exports.CSPM_HOSTS_USAGE = "cspm_hosts_usage";
exports.CUSTOM_TIMESERIES_USAGE = "custom_timeseries_usage";
exports.CUSTOM_TIMESERIES_PERCENTAGE = "custom_timeseries_percentage";
exports.CWS_CONTAINERS_PERCENTAGE = "cws_containers_percentage";
exports.CWS_CONTAINERS_USAGE = "cws_containers_usage";
exports.CWS_HOSTS_PERCENTAGE = "cws_hosts_percentage";
exports.CWS_HOSTS_USAGE = "cws_hosts_usage";
exports.DBM_HOSTS_PERCENTAGE = "dbm_hosts_percentage";
exports.DBM_HOSTS_USAGE = "dbm_hosts_usage";
exports.DBM_QUERIES_PERCENTAGE = "dbm_queries_percentage";
exports.DBM_QUERIES_USAGE = "dbm_queries_usage";
exports.ESTIMATED_INDEXED_LOGS_USAGE = "estimated_indexed_logs_usage";
exports.ESTIMATED_INDEXED_LOGS_PERCENTAGE = "estimated_indexed_logs_percentage";
exports.ESTIMATED_INGESTED_LOGS_USAGE = "estimated_ingested_logs_usage";
exports.ESTIMATED_INGESTED_LOGS_PERCENTAGE = "estimated_ingested_logs_percentage";
exports.ESTIMATED_INDEXED_SPANS_USAGE = "estimated_indexed_spans_usage";
exports.ESTIMATED_INDEXED_SPANS_PERCENTAGE = "estimated_indexed_spans_percentage";
exports.ESTIMATED_INGESTED_SPANS_USAGE = "estimated_ingested_spans_usage";
exports.ESTIMATED_INGESTED_SPANS_PERCENTAGE = "estimated_ingested_spans_percentage";
exports.FARGATE_USAGE = "fargate_usage";
exports.FARGATE_PERCENTAGE = "fargate_percentage";
exports.FUNCTIONS_USAGE = "functions_usage";
exports.FUNCTIONS_PERCENTAGE = "functions_percentage";
exports.INFRA_HOST_USAGE = "infra_host_usage";
exports.INFRA_HOST_PERCENTAGE = "infra_host_percentage";
exports.INVOCATIONS_USAGE = "invocations_usage";
exports.INVOCATIONS_PERCENTAGE = "invocations_percentage";
exports.NPM_HOST_USAGE = "npm_host_usage";
exports.NPM_HOST_PERCENTAGE = "npm_host_percentage";
exports.PROFILED_CONTAINER_USAGE = "profiled_container_usage";
exports.PROFILED_CONTAINER_PERCENTAGE = "profiled_container_percentage";
exports.PROFILED_FARGATE_USAGE = "profiled_fargate_usage";
exports.PROFILED_FARGATE_PERCENTAGE = "profiled_fargate_percentage";
exports.PROFILED_HOST_USAGE = "profiled_host_usage";
exports.PROFILED_HOST_PERCENTAGE = "profiled_host_percentage";
exports.SNMP_USAGE = "snmp_usage";
exports.SNMP_PERCENTAGE = "snmp_percentage";
exports.ESTIMATED_RUM_SESSIONS_USAGE = "estimated_rum_sessions_usage";
exports.ESTIMATED_RUM_SESSIONS_PERCENTAGE = "estimated_rum_sessions_percentage";
exports.ALL = "*";
