// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsStep.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsStep = void 0;
/**
 * The steps used in a Synthetics browser test.
 */
class SyntheticsStep {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsStep.attributeTypeMap;
    }
}
exports.SyntheticsStep = SyntheticsStep;
/**
 * @ignore
 */
SyntheticsStep.attributeTypeMap = {
    allowFailure: {
        baseName: "allowFailure",
        type: "boolean",
    },
    isCritical: {
        baseName: "isCritical",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    noScreenshot: {
        baseName: "noScreenshot",
        type: "boolean",
    },
    params: {
        baseName: "params",
        type: "any",
    },
    timeout: {
        baseName: "timeout",
        type: "number",
        format: "int64",
    },
    type: {
        baseName: "type",
        type: "SyntheticsStepType",
    },
};
