// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsBrowserVariable.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsBrowserVariable = void 0;
/**
 * Object defining a variable that can be used in your browser test.
 * Learn more in the [Browser test Actions documentation](https://docs.datadoghq.com/synthetics/browser_tests/actions#variable).
 */
class SyntheticsBrowserVariable {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsBrowserVariable.attributeTypeMap;
    }
}
exports.SyntheticsBrowserVariable = SyntheticsBrowserVariable;
/**
 * @ignore
 */
SyntheticsBrowserVariable.attributeTypeMap = {
    example: {
        baseName: "example",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    pattern: {
        baseName: "pattern",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "SyntheticsBrowserVariableType",
        required: true,
    },
};
