// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/MetricBulkTagConfigCreateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricBulkTagConfigCreateAttributes = void 0;
/**
 * Optional parameters for bulk creating metric tag configurations.
 */
class MetricBulkTagConfigCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MetricBulkTagConfigCreateAttributes.attributeTypeMap;
    }
}
exports.MetricBulkTagConfigCreateAttributes = MetricBulkTagConfigCreateAttributes;
/**
 * @ignore
 */
MetricBulkTagConfigCreateAttributes.attributeTypeMap = {
    emails: {
        baseName: "emails",
        type: "Array<string>",
        format: "email",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
};
