// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringStandardRuleQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringStandardRuleQuery = void 0;
/**
 * Query for matching rule.
 */
class SecurityMonitoringStandardRuleQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringStandardRuleQuery.attributeTypeMap;
    }
}
exports.SecurityMonitoringStandardRuleQuery = SecurityMonitoringStandardRuleQuery;
/**
 * @ignore
 */
SecurityMonitoringStandardRuleQuery.attributeTypeMap = {
    aggregation: {
        baseName: "aggregation",
        type: "SecurityMonitoringRuleQueryAggregation",
    },
    distinctFields: {
        baseName: "distinctFields",
        type: "Array<string>",
    },
    groupByFields: {
        baseName: "groupByFields",
        type: "Array<string>",
    },
    metric: {
        baseName: "metric",
        type: "string",
    },
    metrics: {
        baseName: "metrics",
        type: "Array<string>",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    query: {
        baseName: "query",
        type: "string",
        required: true,
    },
};
