// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CloudflareAccountUpdateRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudflareAccountUpdateRequestAttributes = void 0;
/**
 * Attributes object for updating a Cloudflare account.
 */
class CloudflareAccountUpdateRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CloudflareAccountUpdateRequestAttributes.attributeTypeMap;
    }
}
exports.CloudflareAccountUpdateRequestAttributes = CloudflareAccountUpdateRequestAttributes;
/**
 * @ignore
 */
CloudflareAccountUpdateRequestAttributes.attributeTypeMap = {
    apiKey: {
        baseName: "api_key",
        type: "string",
        required: true,
    },
    email: {
        baseName: "email",
        type: "string",
    },
};
