// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentIntegrationMetadataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentIntegrationMetadataAttributes = void 0;
/**
 * Incident integration metadata's attributes for a create request.
 */
class IncidentIntegrationMetadataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentIntegrationMetadataAttributes.attributeTypeMap;
    }
}
exports.IncidentIntegrationMetadataAttributes = IncidentIntegrationMetadataAttributes;
/**
 * @ignore
 */
IncidentIntegrationMetadataAttributes.attributeTypeMap = {
    incidentId: {
        baseName: "incident_id",
        type: "string",
    },
    integrationType: {
        baseName: "integration_type",
        type: "number",
        required: true,
        format: "int32",
    },
    metadata: {
        baseName: "metadata",
        type: "IncidentIntegrationMetadataMetadata",
        required: true,
    },
    status: {
        baseName: "status",
        type: "number",
        format: "int32",
    },
};
