// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentCreateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentCreateAttributes = void 0;
/**
 * The incident's attributes for a create request.
 */
class IncidentCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentCreateAttributes.attributeTypeMap;
    }
}
exports.IncidentCreateAttributes = IncidentCreateAttributes;
/**
 * @ignore
 */
IncidentCreateAttributes.attributeTypeMap = {
    customerImpacted: {
        baseName: "customer_impacted",
        type: "boolean",
        required: true,
    },
    fields: {
        baseName: "fields",
        type: "{ [key: string]: IncidentFieldAttributes; }",
    },
    initialCells: {
        baseName: "initial_cells",
        type: "Array<IncidentTimelineCellCreateAttributes>",
    },
    notificationHandles: {
        baseName: "notification_handles",
        type: "Array<IncidentNotificationHandle>",
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
};
