// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TeamUpdateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamUpdateAttributes = void 0;
/**
 * Team update attributes
 */
class TeamUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TeamUpdateAttributes.attributeTypeMap;
    }
}
exports.TeamUpdateAttributes = TeamUpdateAttributes;
/**
 * @ignore
 */
TeamUpdateAttributes.attributeTypeMap = {
    color: {
        baseName: "color",
        type: "number",
        format: "int32",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    handle: {
        baseName: "handle",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
};
