// @aws-sdk/client-scheduler/dist-cjs/Scheduler.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Scheduler = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const CreateScheduleCommand_1 = require("./commands/CreateScheduleCommand");
const CreateScheduleGroupCommand_1 = require("./commands/CreateScheduleGroupCommand");
const DeleteScheduleCommand_1 = require("./commands/DeleteScheduleCommand");
const DeleteScheduleGroupCommand_1 = require("./commands/DeleteScheduleGroupCommand");
const GetScheduleCommand_1 = require("./commands/GetScheduleCommand");
const GetScheduleGroupCommand_1 = require("./commands/GetScheduleGroupCommand");
const ListScheduleGroupsCommand_1 = require("./commands/ListScheduleGroupsCommand");
const ListSchedulesCommand_1 = require("./commands/ListSchedulesCommand");
const ListTagsForResourceCommand_1 = require("./commands/ListTagsForResourceCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const UpdateScheduleCommand_1 = require("./commands/UpdateScheduleCommand");
const SchedulerClient_1 = require("./SchedulerClient");
const commands = {
    CreateScheduleCommand: CreateScheduleCommand_1.CreateScheduleCommand,
    CreateScheduleGroupCommand: CreateScheduleGroupCommand_1.CreateScheduleGroupCommand,
    DeleteScheduleCommand: DeleteScheduleCommand_1.DeleteScheduleCommand,
    DeleteScheduleGroupCommand: DeleteScheduleGroupCommand_1.DeleteScheduleGroupCommand,
    GetScheduleCommand: GetScheduleCommand_1.GetScheduleCommand,
    GetScheduleGroupCommand: GetScheduleGroupCommand_1.GetScheduleGroupCommand,
    ListScheduleGroupsCommand: ListScheduleGroupsCommand_1.ListScheduleGroupsCommand,
    ListSchedulesCommand: ListSchedulesCommand_1.ListSchedulesCommand,
    ListTagsForResourceCommand: ListTagsForResourceCommand_1.ListTagsForResourceCommand,
    TagResourceCommand: TagResourceCommand_1.TagResourceCommand,
    UntagResourceCommand: UntagResourceCommand_1.UntagResourceCommand,
    UpdateScheduleCommand: UpdateScheduleCommand_1.UpdateScheduleCommand,
};
class Scheduler extends SchedulerClient_1.SchedulerClient {
}
exports.Scheduler = Scheduler;
(0, smithy_client_1.createAggregatedClient)(commands, Scheduler);
