// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/LogsIndexUpdateRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogsIndexUpdateRequest = void 0;
/**
 * Object for updating a Datadog Log index.
 */
class LogsIndexUpdateRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LogsIndexUpdateRequest.attributeTypeMap;
    }
}
exports.LogsIndexUpdateRequest = LogsIndexUpdateRequest;
/**
 * @ignore
 */
LogsIndexUpdateRequest.attributeTypeMap = {
    dailyLimit: {
        baseName: "daily_limit",
        type: "number",
        format: "int64",
    },
    disableDailyLimit: {
        baseName: "disable_daily_limit",
        type: "boolean",
    },
    exclusionFilters: {
        baseName: "exclusion_filters",
        type: "Array<LogsExclusion>",
    },
    filter: {
        baseName: "filter",
        type: "LogsFilter",
        required: true,
    },
    numRetentionDays: {
        baseName: "num_retention_days",
        type: "number",
        format: "int64",
    },
};
