// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/LogsIndex.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogsIndex = void 0;
/**
 * Object describing a Datadog Log index.
 */
class LogsIndex {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LogsIndex.attributeTypeMap;
    }
}
exports.LogsIndex = LogsIndex;
/**
 * @ignore
 */
LogsIndex.attributeTypeMap = {
    dailyLimit: {
        baseName: "daily_limit",
        type: "number",
        format: "int64",
    },
    exclusionFilters: {
        baseName: "exclusion_filters",
        type: "Array<LogsExclusion>",
    },
    filter: {
        baseName: "filter",
        type: "LogsFilter",
        required: true,
    },
    isRateLimited: {
        baseName: "is_rate_limited",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    numRetentionDays: {
        baseName: "num_retention_days",
        type: "number",
        format: "int64",
    },
};
