// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsAPIStep.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsAPIStep = void 0;
/**
 * The steps used in a Synthetics multistep API test.
 */
class SyntheticsAPIStep {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsAPIStep.attributeTypeMap;
    }
}
exports.SyntheticsAPIStep = SyntheticsAPIStep;
/**
 * @ignore
 */
SyntheticsAPIStep.attributeTypeMap = {
    allowFailure: {
        baseName: "allowFailure",
        type: "boolean",
    },
    assertions: {
        baseName: "assertions",
        type: "Array<SyntheticsAssertion>",
        required: true,
    },
    extractedValues: {
        baseName: "extractedValues",
        type: "Array<SyntheticsParsingOptions>",
    },
    isCritical: {
        baseName: "isCritical",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    request: {
        baseName: "request",
        type: "SyntheticsTestRequest",
        required: true,
    },
    retry: {
        baseName: "retry",
        type: "SyntheticsTestOptionsRetry",
    },
    subtype: {
        baseName: "subtype",
        type: "SyntheticsAPIStepSubtype",
        required: true,
    },
};
