// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/UsageAttributionPagination.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageAttributionPagination = void 0;
/**
 * The metadata for the current pagination.
 */
class UsageAttributionPagination {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UsageAttributionPagination.attributeTypeMap;
    }
}
exports.UsageAttributionPagination = UsageAttributionPagination;
/**
 * @ignore
 */
UsageAttributionPagination.attributeTypeMap = {
    limit: {
        baseName: "limit",
        type: "number",
        format: "int64",
    },
    offset: {
        baseName: "offset",
        type: "number",
        format: "int64",
    },
    sortDirection: {
        baseName: "sort_direction",
        type: "string",
    },
    sortName: {
        baseName: "sort_name",
        type: "string",
    },
    totalNumberOfRecords: {
        baseName: "total_number_of_records",
        type: "number",
        format: "int64",
    },
};
