// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/UsageAttributionBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageAttributionBody = void 0;
/**
 * Usage Summary by tag for a given organization.
 */
class UsageAttributionBody {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UsageAttributionBody.attributeTypeMap;
    }
}
exports.UsageAttributionBody = UsageAttributionBody;
/**
 * @ignore
 */
UsageAttributionBody.attributeTypeMap = {
    month: {
        baseName: "month",
        type: "Date",
        format: "date-time",
    },
    orgName: {
        baseName: "org_name",
        type: "string",
    },
    publicId: {
        baseName: "public_id",
        type: "string",
    },
    tagConfigSource: {
        baseName: "tag_config_source",
        type: "string",
    },
    tags: {
        baseName: "tags",
        type: "{ [key: string]: Array<string>; }",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "string",
    },
    values: {
        baseName: "values",
        type: "UsageAttributionValues",
    },
};
